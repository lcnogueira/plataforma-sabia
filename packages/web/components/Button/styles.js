/* eslint-disable no-nested-ternary */
import styled, { css } from 'styled-components';

const StyledButton = styled.button`
	${({ theme: { metrics, screens }, bgColor, color, uppercase, disabled }) => css`
		background-color: ${bgColor};
		color: ${color};
		border-radius: ${metrics.baseRadius}rem;
		border: none;
		font-size: 2.2rem;
		text-transform: ${uppercase ? 'uppercase' : 'none'};
		padding: 1.8rem 6rem;
		text-align: center;
		text-decoration: none;
		display: inline-block;

		:hover {
			opacity: 0.8;
		}

		${disabled &&
			css`
				opacity: 0.8;
				cursor: not-allowed;
			`}

		@media (max-width: ${screens.medium}px) {
			font-size: 1.6rem;
			padding: 1.4rem 6rem;
		}
	`}
`;

export const CircularButton = styled.button`
	align-items: center;
	justify-content: center;
	background-color: ${({ bgColor }) => bgColor};
	color: ${({ color }) => color};
	border-radius: 100%;
	height: ${({ height }) => (height ? `${height}rem` : '100%')};
	${({ width }) => (width ? `width: ${width}rem` : '')};
	border: none;
	font-size: ${({ size }) =>
		size === 'small' ? '1.2rem' : size === 'medium' ? '1.6rem' : '2rem'};
	text-transform: uppercase;
	text-align: center;
	text-decoration: none;
	display: inline-block;
	padding: ${({ size }) => (size === 'small' ? '0.2rem !important' : '1rem !important')};

	float: ${({ float }) => float || 'right'};

	:hover {
		opacity: 0.8;
	}

	display: flex;
	position: relative;
`;

const iconButtonModifiers = {
	gray: (colors) => css`
		border: 2px solid ${colors.lightGray2};
		color: ${colors.lightGray2};

		:hover:not(:disabled) {
			color: ${colors.white};
			background-color: ${colors.lightGray2};
		}
	`,
	info: (colors) => css`
		border: 2px solid ${colors.blue};
		color: ${colors.blue};

		:hover:not(:disabled) {
			color: ${colors.white};
			background-color: ${colors.blue};
		}
	`,
	remove: (colors) => css`
		border: 2px solid ${colors.red};
		color: ${colors.red};

		:hover:not(:disabled) {
			color: ${colors.white};
			background-color: ${colors.red};
		}
	`,
	success: (colors) => css`
		border: 2px solid ${colors.secondary};
		color: ${colors.secondary};

		:hover:not(:disabled) {
			color: ${colors.white};
			background-color: ${colors.secondary};
		}
	`,
};

export const IconButton = styled.button`
	${({ theme: { colors }, variant }) => css`
		border: none;
		background: none;
		outline: none;

		display: flex;
		padding: 0.8rem;

		:focus {
			box-shadow: 0px 0px 4px 2px ${colors.primary};
		}

		:disabled {
			cursor: not-allowed;
			opacity: 0.4;
		}

		${!!variant && iconButtonModifiers[variant](colors)}
	`}
`;

const rectangularColorsToTheme = {
	grey: 'lightGray2',
	red: 'red',
	orange: 'primary',
	green: 'darkGreen',
};

const rectangularButtonVariants = {
	text: (colors, colorVariant) => css`
		:hover:not(:disabled) {
			border-color: ${colors[rectangularColorsToTheme[colorVariant]]};
			color: ${colors[rectangularColorsToTheme[colorVariant]]};
			background-color: transparent;
		}
	`,
	outlined: (colors, colorVariant) => css`
		border-color: ${colors[rectangularColorsToTheme[colorVariant]]};
	`,
	filled: (colors, colorVariant) => css`
		color: ${colors.white};
		background-color: ${colors[rectangularColorsToTheme[colorVariant]]};

		:hover:not(:disabled) {
			opacity: 0.74;
		}
	`,
};

export const RectangularButton = styled.button`
	${({ theme: { colors, screens }, variant, colorVariant }) => css`
		border: 2px solid transparent;
		background: none;
		font-size: 1.4rem;
		font-weight: bold;
		width: 100%;

		display: flex;
		align-items: center;
		justify-content: center;

		padding: 0.4rem 0.8rem;
		line-height: 2.4rem;
		text-transform: uppercase;

		&:disabled {
			opacity: 0.5;
			cursor: not-allowed;
		}

		@media screen and (min-width: ${screens.medium}px) {
			width: auto;
		}

		color: ${colors[rectangularColorsToTheme[colorVariant]]};

		:hover:not(:disabled) {
			color: ${colors.white};
			background-color: ${colors[rectangularColorsToTheme[colorVariant]]};
			border-color: ${colors[rectangularColorsToTheme[colorVariant]]};
		}

		:focus {
			box-shadow: 0px 0px 4px 2px ${colors.primary};
		}

		${!!variant && rectangularButtonVariants[variant](colors, colorVariant)};
	`}
`;

export default StyledButton;
