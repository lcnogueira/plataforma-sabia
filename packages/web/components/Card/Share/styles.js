import styled, { css, keyframes } from 'styled-components';
import { AiOutlineShareAlt } from 'react-icons/ai';

const zoom = keyframes`
	0%, 100% {
		transform: scale(1);
	}
	50% {
		transform: scale(1.1);
	}
`;

export const Container = styled.button`
	${({ theme: { sizes } }) => css`
		display: flex;
		align-items: center;
		justify-content: space-between;
		padding: 0.5rem;
		gap: 0.6rem;
		cursor: pointer;
		background: transparent;
		border: none;

		svg {
			width: ${sizes.defaultIcon}rem;
			height: ${sizes.defaultIcon}rem;

			&:hover {
				animation: ${zoom} 1.5s;
			}
		}
	`}
`;

export const Icon = styled(AiOutlineShareAlt).attrs(({ theme: { colors } }) => ({
	color: colors.lightGray2,
}))``;
