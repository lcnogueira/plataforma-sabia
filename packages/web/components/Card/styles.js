import styled, { css } from 'styled-components';

export const CardContainer = styled.div`
	${({ theme: { colors, metrics } }) => css`
		border-radius: ${metrics.baseRadius}rem;

		display: flex;
		flex-direction: column;
		background-color: ${colors.white};
		transition: all 0.3s ease 0s;
		padding: 0.8rem;

		:hover {
			transform: translateY(-0.7rem);
		}
	`}
`;

export const ImageContainer = styled.div`
	${({ theme: { metrics } }) => css`
		border-radius: ${metrics.baseRadius}rem;
		width: 100%;

		overflow: hidden;
		position: relative;
	`}
`;

export const Badge = styled.div`
	position: absolute;
	left: 0;
	padding: 0.4rem 0.8rem;
	font-size: 1.4rem;
	line-height: 2.4rem;
	border-top-right-radius: ${({ theme }) => theme.metrics.baseRadius}rem;
	border-bottom-right-radius: ${({ theme }) => theme.metrics.baseRadius}rem;

	${(props) =>
		props.bottom &&
		css`
			bottom: 2rem;
			background-color: ${({ theme }) => theme.colors.white};
			color: ${({ theme }) => theme.colors.silver};
		`}

	${(props) =>
		props.top &&
		css`
			top: 2rem;
			background-color: ${({ theme }) => theme.colors.white};
			color: ${({ theme }) => theme.colors.black};
		`}
`;

export const Content = styled.div`
	display: flex;
	flex-direction: column;
	margin-top: 1.6rem;
	font-weight: 400;
`;

export const MainTitle = styled.h3`
	font-family: 'Montserrat';
	font-weight: 500;
	font-size: 1.6rem;
	margin-bottom: 3rem;
	line-height: 2.4rem;
`;

export const TextContainer = styled.div`
	display: flex;
	align-items: center;
	justify-content: space-between;
	font-size: 1.2rem;
	padding-top: 0.8rem;
	border-top: 0.1rem solid ${({ theme }) => theme.colors.border};
`;

export const InstitutionText = styled.span`
	${({ theme: { colors } }) => css`
		display: flex;
		align-items: center;
		justify-content: center;
		font-weight: 500;
		color: ${colors.lightGray2};
		text-transform: uppercase;

		> svg {
			path {
				stroke: ${colors.lightGray2};
				stroke-width: 1.5;
			}
			margin-right: 0.8rem;
		}
	`}
`;

export const CalendarText = styled.div`
	display: flex;
	align-items: center;

	span {
		color: ${({ theme }) => theme.colors.secondary};
		margin-left: 0.5rem;
		display: block;
	}

	svg {
		width: ${({ theme }) => theme.sizes.smallIcon}rem;
		height: ${({ theme }) => theme.sizes.smallIcon}rem;
	}
`;

export const LikesWrapper = styled.div`
	margin-bottom: 1rem;
	> div {
		padding: 0;
	}
`;

export const Price = styled.h2`
	${({ theme: { colors } }) => css`
		font-size: 2.8rem;
		line-height: 3.3rem;
		color: ${colors.secondary};
		margin-bottom: 1.2rem;
	`}
`;
