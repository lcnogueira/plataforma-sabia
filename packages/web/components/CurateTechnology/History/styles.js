import styled, { css } from 'styled-components';

export const CommentTitle = styled.div`
	${({ theme: { colors } }) => css`
		display: flex;
		align-items: baseline;
		font-weight: 700;
		color: ${colors.black};

		> span {
			font-size: 1.4rem;
			font-weight: 500;
			margin-left: 0.5rem;
		}
	`}
`;

export const Comment = styled.div`
	${({ theme: { colors, metrics } }) => css`
		background-color: ${colors.lightGray4};
		border-radius: ${metrics.baseRadius}rem;
		padding: 0.8rem;
		display: flex;
		flex-direction: column;
		flex-basis: 100%;
	`}
`;

export const CommentContent = styled.div`
	${({ theme: { colors, metrics } }) => css`
		background-color: ${colors.white};
		border-radius: ${metrics.baseRadius}rem;
		margin-top: 0.8rem;
		padding: 0.8rem;
		height: 100%;

		> span {
			color: ${colors.lightGray2};
			font-size: 1.2rem;
		}
	`}
`;

export const CommentText = styled.div`
	${({ theme: { colors } }) => css`
		color: ${colors.black};
		font-size: 1.4rem;
		line-height: 2.4rem;
		margin-top: 0.8rem;

		ol,
		ul {
			padding-left: 1.4rem;
		}
	`}
`;
