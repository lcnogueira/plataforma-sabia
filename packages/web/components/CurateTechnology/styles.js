import styled, { css } from 'styled-components';
import { TabList as RTablList } from 'react-tabs';

import { Tabs, Tab as RTab } from '../Tab';

export const Wrapper = styled(Tabs)`
	display: flex;
	flex-direction: column;
	flex-grow: 1;
`;

export const TitleWrapper = styled.div`
	display: flex;
	align-items: center;

	div:first-child {
		margin-left: 1.4rem;
	}
`;

export const Tab = styled(RTab)`
	${({ selected, theme: { colors } }) => css`
		display: flex;
		align-items: center;
		justify-content: center;
		text-transform: uppercase;

		border: none;
		bottom: 0;
		position: relative;

		font-weight: 700;
		font-size: 1.2rem;

		background-color: transparent;
		color: ${selected ? colors.secondary : colors.lightGray2};

		padding: 0.8rem;

		:after {
			content: '';
			position: absolute;
			bottom: -2px;
			right: 0;
			left: 0;
			border-bottom: 2px solid ${selected ? colors.secondary : colors.lightGray4};
		}
	`}
`;

export const TabsHeader = styled.div`
	${({ theme: { colors } }) => css`
		border-bottom: 2px solid ${colors.lightGray4};
		margin-top: 1.6rem;
	`}
`;

export const TabList = styled(RTablList)`
	${({ theme: { screens } }) => css`
		padding: 0;
		display: grid;

		@media screen and (min-width: ${screens.medium}px) {
			grid-template-columns: 1fr 1fr;
			grid-template-rows: 1fr 1fr 1fr;
		}

		@media screen and (min-width: ${screens.large}px) {
			grid-template-columns: 1fr 1fr 1fr 1fr;
			grid-template-rows: 1fr 1fr;
		}

		@media screen and (min-width: 1280px) {
			grid-template-columns: repeat(6, 1fr);
			grid-template-rows: 1fr;
		}
	`}
`;

export const Footer = styled.footer`
	${({ theme: { screens } }) => css`
		display: flex;
		flex-flow: row wrap;

		button {
			margin-top: 1.4rem;
		}

		> button {
			margin-right: 2.2rem;
		}

		> div {
			display: flex;
			flex-flow: row wrap;
			flex-grow: 1;
		}

		@media screen and (min-width: ${screens.medium}px) {
			button:not(:first-child) {
				margin-left: 2.2rem;
			}

			> div {
				margin-left: auto;
				justify-content: flex-end;
			}
		}
	`}
`;

/** *
 * Shared between tabs
 ** */

export const Container = styled.div`
	${({ theme: { screens }, flexDirection }) => css`
		display: flex;
		flex-direction: ${flexDirection};
		padding: 3.2rem 0;
		max-width: 116.4rem;

		@media screen and (max-width: ${screens.medium}px) {
			flex-direction: column;
		}
	`}
`;

export const ContentBox = styled.div`
	${({ theme: { colors, screens }, flexBasis }) => css`
		flex-basis: ${flexBasis || '50%'};

		> div:not(:first-child) {
			margin-top: 1.6rem;
		}

		@media screen and (max-width: ${screens.medium}px) {
			flex-basis: 100%;

			:last-child > div:first-child {
				margin-top: 1.6rem;
			}
		}

		@media screen and (min-width: ${screens.medium}px) {
			:not(:first-child) {
				margin-left: 0.8rem;
				padding-left: 0.8rem;
				border-left: 1px solid ${colors.lightGray4};
			}
		}
	`}
`;
