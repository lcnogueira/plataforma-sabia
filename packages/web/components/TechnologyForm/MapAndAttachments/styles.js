import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
	margin-bottom: 4rem;
`;

export const UploadBox = styled.div`
	flex-direction: row;
	display: flex;
	margin: 0;
	margin-bottom: 1rem;
	border: 1px solid ${({ theme }) => theme.colors.lightGray3};
	padding: 1rem;
	cursor: pointer;

	&:hover {
		background-color: ${({ theme }) => theme.colors.lightGray3};
	}
`;

export const IconLink = styled.a`
	color: ${({ theme }) => theme.colors.darkGray};
`;

export const Place = styled.div``;

export const Div = styled.div``;

export const IconBox = styled.div`
	display: flex;
	justify-content: center;
	align-items: center;

	& > * {
		margin: 0.25rem;
	}
`;

export const IconTextBox = styled.div`
	font-size: 1.2rem;
	font-weight: 300;
	display: flex;
	justify-content: flex-start;
	align-items: flex-start;
	flex-direction: column;
`;

export const InputTextBox = styled.div`
	flex: 3;
	margin: auto 0;
`;

export const UploadedDocuments = styled.div`
	display: flex;
	flex-direction: column;
`;

export const UploadedImages = styled.div`
	display: grid;
	grid-template-columns: repeat(auto-fill, 30%);
	grid-gap: 1rem;

	@media (max-width: ${({ theme }) => theme.screens.small}px) {
		grid-template-columns: repeat(auto-fill, 100%);
	}
`;

export const Suggestion = styled.div`
	margin: 0.5rem 0;
	padding: 1rem;
	width: 100%;
	font-size: 1.4rem;
	cursor: pointer;
`;

export const GoogleAddressSugestions = styled.div`
	display: flex;
	flex-direction: column;
`;

export const SvgBox = styled.div`
	text-align: center;
	flex: 1;
	margin: auto 0;
`;

export const Title = styled.h3``;

export const Media = styled.img`
	width: 100%;
	filter: opacity(0.7);
`;

export const IconRow = styled.div`
	${({ row, alignItems }) => css`
		display: flex;
		flex-direction: ${row ? 'row' : 'column'};

		& > * {
			margin-bottom: 1rem;
		}

		& > a {
			padding: 0;
			margin-right: 1rem;
			display: flex;
		}

		align-items: ${alignItems || 'center'};
	`}
`;

export const Button = styled.button.attrs(() => ({ type: 'button' }))`
	${({ theme: { colors } }) => css`
		border: none;
		outline: none;

		display: flex;
		align-items: center;

		padding: 0.2rem;
		padding-left: 0.6rem !important;
		padding-right: 0.6rem !important;
		line-height: 2.4rem;

		text-transform: uppercase;
		font-size: 1.4rem;
		font-weight: 700;
		color: ${colors.red};
		border: 2px solid ${colors.red};

		> svg {
			margin-right: 0.4rem;
		}

		:hover,
		:focus {
			color: ${colors.white};
			background: ${colors.red};
		}
	`}
`;

export const RadioWrapper = styled.div`
	${({ theme: { colors } }) => css`
		display: flex;
		align-items: center;
		white-space: nowrap;

		> input {
			appearance: none;
		}
		> input:checked + label {
			color: ${colors.white};
			background-color: ${colors.secondary};
			border: 2px solid ${colors.secondary};
		}
		> label {
			display: flex;
			align-items: center;
			color: ${colors.lightGray2};
			text-align: center;
			padding: 0.2rem 0.6rem;
			cursor: pointer;
			font-size: 1.4rem;
			font-weight: 700;
			line-height: 2.4rem;
			border: 2px solid ${colors.lightGray2};
			text-transform: uppercase;

			> svg {
				margin-right: 0.4rem;
			}
		}
		:focus-within > label {
			background-color: ${colors.secondary};
		}
	`}
`;

export const InputVideoWrapper = styled.section`
	width: 100%;
	display: flex;
	align-items: center;
	margin: 0 -8px 16px;

	> div {
		flex: 1;
		margin-right: 10px;
	}
`;

const VideoContainer = styled.section`
	background: ${({ theme: { colors } }) => colors.white};
	width: 100%;
	padding: 16px 32px 32px;
	margin-top: 8px;
	margin-bottom: 16px;
`;

const VideosWrapper = styled.section`
	border-top: 1px solid ${({ theme: { colors } }) => colors.lightGray4};
	border-bottom: 1px solid ${({ theme: { colors } }) => colors.lightGray4};
	padding: 16px 16px 0;
	margin: 0 -32px;
	display: flex;
	flex-wrap: wrap;
	align-items: center;
	flex-wrap: wrap;
`;

const VideoItem = styled.section`
	width: 130px;
	display: flex;
	flex-direction: column;
	margin: 0 8px 16px;

	a {
		display: block;
		padding: 0;
		margin-bottom: 10px;

		&:hover {
			opacity: 0.7;
		}
	}

	img {
		background: ${({ theme: { colors } }) => colors.lightGray2};
		width: 100%;
		display: block;
	}
`;

const RemoveVideoButton = styled.button`
	display: flex;
	align-items: center;
	align-self: center;
	border: 0;
	outline: none;

	text-transform: uppercase;
	font-weight: bold;
	font-size: 1.4rem;
	line-height: 2.4rem;

	background: none;
	color: ${({ theme: { colors } }) => colors.red};
	padding: 0.2rem 0.6rem;

	:hover,
	:focus {
		color: ${({ theme: { colors } }) => colors.white};
		background: ${({ theme: { colors } }) => colors.red};
	}
`;

const EmptyVideos = styled.section`
	text-align: center;
`;

export const Videos = ({ data, onRemove, children }) => (
	<VideoContainer>
		{children}
		{data?.length ? (
			<VideosWrapper>
				{data.map((video, idx) => (
					<VideoItem key={`video_${video.videoId}`}>
						<a
							href={`//www.youtube.com/watch?v=${video.videoId}`}
							target="_blank"
							rel="noreferrer"
						>
							<img src={video.thumbnail} alt={`Youtube video ${video.videoId}`} />
						</a>
						<RemoveVideoButton type="button" onClick={() => onRemove(idx)}>
							Remover
						</RemoveVideoButton>
					</VideoItem>
				))}
			</VideosWrapper>
		) : (
			<EmptyVideos>
				<p>Nenhum video adicionado</p>
			</EmptyVideos>
		)}
	</VideoContainer>
);

Videos.propTypes = {
	data: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
	children: PropTypes.children().isRequired,
	onRemove: PropTypes.func().isRequired,
};
