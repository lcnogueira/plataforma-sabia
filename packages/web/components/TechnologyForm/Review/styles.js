import styled from 'styled-components';
import * as Layout from '../../Common/Layout';

export const Cell = styled(Layout.Cell)`
	width: 100%;
`;

export const Row = styled.div`
	display: flex;
	align-items: flex-start;

	& > *:not(:first-child):not(:last-child) {
		margin: 0 10px;
	}

	@media (max-width: ${({ theme }) => theme.screens.large}px) {
		flex-direction: column;
		margin-top: 1.5rem;
	}
`;

export const Col = styled.div`
	flex: ${({ size }) => size || 1};

	@media (max-width: ${({ theme }) => theme.screens.small}px) {
		width: 100%;
	}
`;

export const Wrapper = styled.div`
	margin-bottom: 4rem;
`;

export const Label = styled.label`
	display: block;
	margin: 1rem 0;
	text-transform: uppercase;

	a {
		padding: 0;
	}

	input[type='checkbox'] {
		margin-right: 1rem;
	}
`;

export const Checkbox = styled.input.attrs(({ name }) => ({
	name,
	id: name,
	type: 'checkbox',
}))`
	background: blue;
	border: 1px solid red;
`;
