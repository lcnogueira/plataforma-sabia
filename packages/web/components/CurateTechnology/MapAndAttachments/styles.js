import styled, { css } from 'styled-components';
import * as MapAndAttachments from '../../TechnologyForm/MapAndAttachments/styles';

export const UploadsTitle = styled.span`
	${({ theme: { colors } }) => css`
		display: block;
		font-weight: 500;
		font-size: 1.4rem;
		margin-bottom: 1rem;
		text-transform: uppercase;
		color: ${colors.lightGray};

		&:not(:first-child) {
			margin-top: 2.4rem;
		}
	`}
`;

export const UploadedDocuments = styled(MapAndAttachments.UploadedDocuments)``;

export const UploadedImages = styled(MapAndAttachments.UploadedImages)`
	grid-template-columns: repeat(auto-fill, 20%);
`;

export const Media = styled(MapAndAttachments.Media)`
	cursor: pointer;
	filter: opacity(1);
`;

export const IconRow = styled(MapAndAttachments.IconRow)``;

export const IconLink = styled(MapAndAttachments.IconLink)``;

export const ImageCaption = styled.p`
	${({ theme: { colors } }) => css`
		font-size: 1.2rem;
		margin-top: -1rem;
		color: ${colors.lightGray};
	`}
`;
