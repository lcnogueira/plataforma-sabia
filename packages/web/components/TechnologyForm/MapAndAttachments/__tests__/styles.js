import React from 'react';
import { render, screen } from 'test-utils';
import { VideosWrapper, VideoItem } from '../styles';

const video = {
	thumbnail: 'http://i3.ytimg.com/vi/8h7p88oySWY/hqdefault.jpg',
	videoId: '8h7p88oySWY',
};

test('it render the <VideosWrapper /> component', () => {
	const { container } = render(<VideosWrapper />);

	expect(container).toMatchSnapshot();
});

test('it render the <VideoItem /> component', async () => {
	const { container } = render(
		<VideoItem>
			<a href={`//www.youtube.com/watch?v=${video.videoId}`} target="_blank" rel="noreferrer">
				<img src={video.thumbnail} alt={`Youtube vÃ­deo ${video.videoId}`} />
			</a>
		</VideoItem>,
	);

	expect(screen.getByRole('img', { src: video.thumbnail })).toBeInTheDocument();
	expect(container).toMatchSnapshot();
});
