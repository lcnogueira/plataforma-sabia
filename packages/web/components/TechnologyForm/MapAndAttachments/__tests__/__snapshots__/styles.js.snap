// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it render the <ButtonVideoAdd /> component 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  border: none;
  outline: none;
  text-transform: uppercase;
  font-weight: bold;
  font-size: 1.4rem;
  line-height: 2.4rem;
  background: none;
  color: hsl(166,100%,34%);
  border: 2px solid hsl(166,100%,34%);
  padding: 0.8rem 0.6rem;
}

.c0:disabled {
  pointer-events: none;
  opacity: 0.5;
}

.c0:hover,
.c0:focus {
  color: hsla(0,0%,100%);
  background: hsl(166,100%,34%);
}

<div>
  <button
    class="c0"
    type="button"
  >
    Adicionar
  </button>
</div>
`;

exports[`it render the <VideoContainer /> component 1`] = `
.c0 {
  background: hsla(0,0%,100%);
  width: 100%;
  padding: 16px 32px 32px;
  margin-top: 8px;
  margin-bottom: 16px;
}

<div>
  <section
    class="c0"
  />
</div>
`;

exports[`it render the <VideoItem /> component 1`] = `
.c0 {
  width: 130px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0 8px 16px;
}

.c0 a {
  display: block;
  padding: 0;
  margin-bottom: 10px;
}

.c0 a:hover {
  opacity: 0.7;
}

.c0 img {
  background: hsla(0,0%,52%);
  width: 100%;
  display: block;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  border: 0;
  outline: none;
  text-transform: uppercase;
  font-weight: bold;
  font-size: 1.4rem;
  line-height: 2.4rem;
  background: none;
  color: hsla(0,79%,63%);
  padding: 0.2rem 0.6rem;
}

.c1:hover,
.c1:focus {
  color: hsla(0,0%,100%);
  background: hsla(0,79%,63%);
}

<div>
  <section
    class="c0"
  >
    <a
      href="//www.youtube.com/watch?v=8h7p88oySWY"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="Youtube vÃ­deo 8h7p88oySWY"
        src="http://i3.ytimg.com/vi/8h7p88oySWY/hqdefault.jpg"
      />
    </a>
    <button
      class="c1"
      type="button"
    >
      Remover
    </button>
  </section>
</div>
`;

exports[`it render the <VideosWrapper /> component 1`] = `
.c0 {
  border-top: 1px solid hsl(0,0%,93%);
  border-bottom: 1px solid hsl(0,0%,93%);
  padding: 16px 16px 0;
  margin: 0 -32px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

<div>
  <section
    class="c0"
  />
</div>
`;
